### DISCUSSION TITLE
presigned URLs from Minio

### DISCUSSION THREAD

**Discussion by man-tax** [Budibase/budibase#5624] [Help] [answered]
Hi everyone,

I was able to successfully get presigned URLs from the Minio API using JS. So I can keep a bucket private but allow users to download their files using those special links that expire ([https://docs.min.io/docs/javascript-client-api-reference.html#presignedGetObject]).

Now I'm wondering how/ where I can get this into budibase? The script includes setting up a Minio Client object and calling the operation which results in the presigned URL.

So something like this

const Minio = require('minio')

var client = new Minio.Client({
    endPoint: '10.0.0.3',
    port: 9000,
    useSSL: false,
    accessKey: 
    secretKey: 

and

client.presignedGetObject('bucket', 'file', (err, presignedUrl)

I'm really not an expert, but understand I can't use the JS scripting automation for this? Minio doesn't provide a REST API as far as I know (or just to run SQL queries). Any help is very much appreciated!

Thanks
Sebastian

**Comment by melohagan** [comment, marked as answer]
Hi @man-tax 

You cannot use async functions, or JS libraries in Budibase at the moment. 

For presigned URL attachments, there is an issue for that here: https://github.com/Budibase/budibase/issues/3582

**Reply by man-tax** [reply]
I've managed to get the link into budibase using an http request to a node.js-app I've set up, yay

![image](https://user-images.githubusercontent.com/104101494/165817036-3051beb8-d8a0-4d97-9ae4-e1d5daec70f9.png)

**Reply by oswaldB** [reply]
Perfect! What is the endpoint you use? Https://domain.com:9000 ? Thank you
